test/integration/index.js

```diff
@@ -0,0 +1,286 @@
+'use strict'
+process.env.INGESTION_KEY = 'abc123'
+
+const os = require('os')
+const nock = require('nock')
+const {test, threw} = require('tap')
+
+const {HOSTNAME_REGEX} = require('../../lib/constants.js')
+const {buildLoggerURL} = require('../../lib/logger.js')
+const config = require('../../lib/config.js')
+const {handler} = require('../../index.js')
+const {s3} = require('../../lib/transformer.js')
+const {
+  formatObjectKey
+, getProperty
+, setProperty
+, trimTags
+} = require('../../lib/utils.js')
+
+const responseText = 'This is the ingester response'
+const BUCKET_NAME = 'random-name'
+const FILE_NAME = 'log+File.json'
+const LOG_LEVEL = 'INFO'
+const LOG_LINE = 'test log'
+const SAMPLE_HOSTNAME = 'sampleHostname/test'
+const SAMPLE_TAGS = ' ,test,sample ,something'
+const TIMESTAMP = Date.now()
+const EVENT_DATA = {
+  Records: [{
+    eventTime: new Date(TIMESTAMP)
+  , s3: {
+      bucket: {
+        name: BUCKET_NAME
+      }
+    , object: {
+        key: FILE_NAME
+      }
+    }
+  }]
+}
+
+// NOTE: We will NOT test the use of ENV vars directly here since they are cumbersome
+// to work with (they can never be updated). For this, we will trust that env-config
+// will do its job and read from the envrionment properly, but we will set the
+// config's keys manually in these tests.
+config.validateEnvVars()
+nock.disableNetConnect()
+
+test('handler is the main method in lambda functions', async (t) => {
+  t.test('test fully', async (t) => {
+    t.on('end', async () => {
+      nock.cleanAll()
+    })
+
+    config.set('tags', SAMPLE_TAGS)
+    const input = JSON.stringify({
+      log: LOG_LINE
+    })
+
+    const getObject = s3.getObject
+    s3.getObject = function({
+      Bucket: BUCKET_NAME
+    , Key: FILE_NAME
+    }) {
+      return {
+        Body: input
+      }
+    }
+
+    nock(buildLoggerURL(config))
+      .post('', (body) => {
+        const numProps = Object.keys(body).length
+        t.strictEqual(numProps, 2, 'Number of request body properties')
```

Comment:
Seems like this test is more for the logger, but it's hard to say what the intention is.  I see there's another comment on this.

Reply:
I'll leave this topic alone now.  I think just making the description a little better would help.